test/integration/spec/room.js

```diff
@@ -713,6 +702,130 @@ describe('Room', function() {
+      const tracksToPublish = Array(roomDetails.max_concurrent_published_tracks - tracksPublishedAlready).fill(0);
+
+      const tracks = await waitFor(tracksToPublish.map(() => createLocalAudioTrack({ fake: true })));
+      await waitFor(bobRoom.localParticipant.publishTracks(tracks), `max Tracks to get published: ${roomSid}`);
+
+      const oneMoreTrack = await createLocalAudioTrack({ fake: true });
+      let publishError = null;
+      try {
+        await bobRoom.localParticipant.publishTrack(oneMoreTrack);
+      } catch (error) {
+        publishError = error;
+        assert(publishError.code === 53203, `Unexpected ErrorCode: ${publishError.code}`);
+      }
+      assert(publishError !== null, 'was expecting an error publishing track');
+    });
+
+    it('emits synthetic participantDisconnected when participant unpublish last track', async () => {
+      const bobDisConnectedPromise = new Promise(resolve => aliceRoom.once('participantDisconnected', resolve));
+
+      [...bobRoom.localParticipant.tracks.values()].forEach(trackPub => trackPub.unpublish());
+      await waitFor(bobDisConnectedPromise, `waiting for synthetic participantDisConnected: ${roomSid}`);
+    });
+  });
+
+  (defaults.topology === 'group' ? describe : describe.skip)('large rooms ParticipantMaxTracksExceededError', () => {
+    [1, 2, 5].forEach(nTracksToPublishLater => {
+      it(`throws ParticipantMaxTracksExceededError when publishing ${nTracksToPublishLater} tracks when they exceed the limit:`, async () => {
+
+        // create a large group room
+        const roomSid = await createRoom(randomName(), defaults.topology, { MaxParticipants: 51 });
+        const roomDetails = await getRoom(roomSid);
+
+        assert(roomDetails.max_concurrent_published_tracks === 16);
+        assert(roomDetails.max_participants === 51);
+
+        // alice join the room
+        const aliceRoom = await connect(getToken('Alice'), Object.assign({ name: roomSid }, defaults, { tracks: [] }));
+
+        // bob joins room with 0 tracks
+        const bobConnectedPromise = new Promise(resolve => aliceRoom.once('participantConnected', resolve));
+        const bobRoom = await connect(getToken('Bob'), Object.assign({ name: roomSid }, defaults, { tracks: [] }));
+
+        // it does not invoke participantConnected.
+        await waitForNot(bobConnectedPromise, `received unexpected participantConnected: ${roomSid}`);
+
+        // bob publishes max_concurrent_published_tracks - nTracksToPublishLater + 1
+        const nTracksToPublishInitially = roomDetails.max_concurrent_published_tracks - nTracksToPublishLater + 1;
+        const tracksToPublishInitially = await waitFor(Array(nTracksToPublishInitially).fill(0).map(() => createLocalAudioTrack({ fake: true })), `creating ${nTracksToPublishInitially} tracks`);
+        await waitFor(bobRoom.localParticipant.publishTracks(tracksToPublishInitially), `initial ${nTracksToPublishInitially} tracks to get published: ${roomSid}`);
```

Comment:
@makarandp0 You can pass the `nTracksToPublishInitially` tracks in ConnectOptions instead of publishing after joining the Room.